File: src/main/java/infinitespire/relics/DarkRift.java
package infinitespire.relics;

import com.megacrit.cardcrawl.actions.animations.VFXAction;
import com.megacrit.cardcrawl.actions.common.MakeTempCardInDrawPileAction;
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.Relic;
import infinitespire.effects.BlackCardEffect;
import infinitespire.helpers.CardHelper;

public class DarkRift extends Relic {

	public static final String ID = InfiniteSpire.createID("DarkRift");

	public DarkRift() {
		super(ID, "darkRift", RelicTier.STARTER, LandingSound.MAGICAL);
	}

	@Override
	public void atBattleStart() {
		this.flash();
		AbstractDungeon.actionManager.addToBottom(new VFXAction(new BlackCardEffect()));
		AbstractDungeon.actionManager.addToBottom(new RelicAboveCreatureAction(AbstractDungeon.player, this));
		AbstractDungeon.actionManager.addToBottom(new MakeTempCardInDrawPileAction(CardHelper.getRandomBlackCard(), 1, true, true));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/DarkRift.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.cards.utils.CardManager
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.vfx.BlackCardVfx
import com.megacrit.cardcrawl.actions.animations.VFXAction
import com.megacrit.cardcrawl.actions.common.MakeTempCardInDrawPileAction
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction

class DarkRift : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "DarkRift".makeID()
    private const val IMG = "darkRift"
    private val TIER = RelicTier.STARTER
    private val SOUND = LandingSound.MAGICAL
  }

  override fun atBattleStart() {
    flash()
    addToBot(VFXAction(BlackCardVfx()))
    addToBot(RelicAboveCreatureAction(player, this))
    addToBot(
      MakeTempCardInDrawPileAction(
        CardManager.getRandomBlackCard(),
        1,
        true,
        true
      )
    )
  }
}